viewing_party/party.py

```diff
@@ -1,23 +1,146 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    if title and genre and rating:
+        movie_dict = {"title": title, "genre": genre, "rating": rating}
+        return movie_dict
+    else:
+        return None
+
+def add_to_watched(user_data, movie):
+    '''
+    Takes two arguments, user_data and movie, where user_data 
+    is a dictionary that stores all of the movies the user has watched 
+    and movie is a dictionary which contains the information for a movie
+    
+    Adds the value of movie to the "watched" list inside of user_data
+
+    Returns user_data
+    '''
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    '''
+    adds movie to the watchlist list inside of user_data,
+    where movie is a dictionary and user_data is a dictionary with a list of dictionaries 
+    '''
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    '''
+    moves title from a user's watchlist to their watched list
+    '''
+    # watchlist_list = user_data["watchlist"]
+    # print(watchlist_list)
+    for index in range(len(user_data["watchlist"])):
+        if user_data["watchlist"][index]["title"] == title:
+            user_data["watched"].append(user_data["watchlist"].pop(index))

```

Comment:
Two notes here, but both of the issues are pretty minor.

While you can use `pop` here, you can also use `remove`, and when combined with not using `range` in the `for` loop you get an arguably cleaner implementation:

```python
for movie in user_data["watchlist"]:
    if movie["title"] == title:
        user_data["watchlist"].remove(movie)
        user_data["watched"].append(movie)
```

Though both this implementation and your implementation are doing something rather dangerous and modifying the list the loop is iterating through. However, I had been allowing this implementation in earlier submissions, so I am going to continue to allow it for the sake of consistency.